library/core/src/array/mod.rs

```diff
@@ -533,11 +533,12 @@ impl<T, const N: usize> [T; N] {
     /// assert_eq!(c, Some(a));
     /// ```
     #[unstable(feature = "array_try_map", issue = "79711")]
-    pub fn try_map<F, R>(self, f: F) -> ChangeOutputType<R, [R::Output; N]>
+    pub fn try_map<R>(
+        self,
+        f: impl FnMut(T) -> R,
+    ) -> <R::Residual as Residual<[R::Output; N]>>::TryType
```

Comment:
This extremely noisy signature cannot reuse `ChangeOutputType` without falling afoul of either the `type_alias_bounds` lint or having an _even more grody_ `<<T as Try>::Residual as Residual<blahblahblah>>::TryType`.

One option that we do have that requires slightly more implementation work (changing the internals of the ones that use NeverShortCircuit, mostly) is this:
```rust
    #[unstable(feature = "array_try_map", issue = "79711")]
    pub fn try_map<R>(
        self,
        f: impl FnMut(T) -> R,
    ) -> impl Try<Output = [R::Output; N], Residual = R::Residual>
    where
        R: Try<Residual: Residual<[R::Output; N]>>,
```

Even further alternates might exist.

Reply:
basically agreed, which is what makes me annoyed the `type_alias_bounds` lint fires on changing the `type ChangeOutputType` alias to match. like, that can't possibly be right, can it?

Reply:
Oh, I'm reciting this issue: https://github.com/rust-lang/rust/issues/125709